common/src/notifications.rs

```diff
@@ -41,3 +116,142 @@ pub fn set_badge(count: u32) -> Result<(), String> {
+fn show_with_action(notification: Notification, action_id: String, action: NotificationAction) {
+    #[cfg(target_os = "windows")]
+    {
+        // Notify-rust does not support windows actions so we use the underlying system directly
+        // See https://gist.github.com/allenbenz/a0fb225aef43df4b1be1c005fb4c2811 for general idea
+        let duration = match notification.timeout {
+            notify_rust::Timeout::Default => "duration=\"short\"",
+            notify_rust::Timeout::Never => "duration=\"long\"",
+            notify_rust::Timeout::Milliseconds(t) => {
+                if t >= 25000 {
+                    "duration=\"long\""
+                } else {
+                    "duration=\"short\""
+                }
+            }
+        };
+        //TODO set proper app id
+        let app_id = POWERSHELL_APP_ID.to_string();
+        let template_binding = "ToastGeneric";
+        let actions = format!(
+            r#"<action content="{}" arguments="{}"/>"#,
+            &get_local_text(&action_id),
+            &action_id
+        );
+
+        let toast_xml = windows::Data::Xml::Dom::XmlDocument::new().unwrap();
+        toast_xml
+            .LoadXml(&windows::runtime::HSTRING::from(format!(
+                "<toast {} {}>
+                    <visual>
+                        <binding template=\"{}\">
+                        {}
+                        {}{}{}
+                        </binding>
+                    </visual>
+                    {}
+                    <actions>
+                        {}
+                    </actions>
+                </toast>",
+                duration,
+                String::new(), //Scenario
+                template_binding,
+                &notification.icon,
+                &notification.summary,
+                notification.subtitle.as_ref().map_or("", AsRef::as_ref),
+                &notification.body,
+                r#"<audio silent='true'/>"#, //Already handled in uplink
+                actions
+            )))
+            .unwrap();
+
+        // Create the toast
+        let toast_notification =
+            windows::UI::Notifications::ToastNotification::CreateToastNotification(&toast_xml)
+                .unwrap();
+
+        toast_notification
+            .Activated(windows::Foundation::TypedEventHandler::new(
+                move |_sender, result: &Option<windows::runtime::IInspectable>| {
+                    let event: Option<
+                        windows::runtime::Result<
+                            windows::UI::Notifications::ToastActivatedEventArgs,
+                        >,
+                    > = result.as_ref().map(windows::runtime::Interface::cast);
+                    let arguments = event
+                        .and_then(|val| val.ok())
+                        .and_then(|args| args.Arguments().ok());
+                    if let Some(val) = arguments {
```

Comment:
problem is on linux: clicking on the notification sends the same action string `__closed` as just letting the notification fade out.  there is no distinction between them so we need an action string. so while for windows and mac it might be possible without action string the method args will still need to be like this.
the action also makes it so you need to click on a button in the notification to make it do its thing, else it just brings the app into the front (which might or might not be desirable. personally am split about this)

Reply:
ok, that's fine. But can  `action_id` be generated from `NotificationAction`? 
ex:
```
fn show_with_action(notification: Notification, action: NotificationAction)  {
    let action_id = format!("{}", action);
    ...
```